server/src/services/documentSymbol/onDocumentSymbol.ts

```diff
@@ -71,50 +71,100 @@ export function onDocumentSymbol(serverState: ServerState) {
 
         const builder = new SymbolTreeBuilder();
 
-        const visitors: SymbolVisitor[] = [
-          new StructDefinition(document, builder),
-          new StructMember(document, builder),
-          new InterfaceDefinition(document, builder),
-          new FunctionDefinition(document, builder),
-          new ContractDefinition(document, builder),
-          new EventDefinition(document, builder),
-          new StateVariableDeclaration(document, builder),
-          new VariableDeclaration(document, builder),
-          new ConstantDefinition(document, builder),
-          new ConstructorDefinition(document, builder),
-          new EnumDefinition(document, builder),
-          new ErrorDefinition(document, builder),
-          new FallbackFunctionDefinition(document, builder),
-          new LibraryDefinition(document, builder),
-          new ModifierDefinition(document, builder),
-          new ReceiveFunctionDefinition(document, builder),
-          new UserDefinedValueTypeDefinition(document, builder),
-          new YulFunctionDefinition(document, builder),
-          new UnnamedFunctionDefinition(document, builder),
+        const finders: SymbolFinder[] = [
```

Comment:
I suggest creating/caching the `finders` array once in the beginning here in this module, instead of recreating it on every feature invocation, so that the queries don't have to be parsed/validated every single call.

Maybe each `SymbolFinder` can just create the query directly in its constructor and store it, instead of exposing its source code as a public string?

- Same thing fo semantic highlighting.

Reply:
I'm happy to address this later. No need to block this. Thanks.